### ISSUE TITLE
Menu component: Let consumer add SearchInput/TextInput and remove hasSearchInput prop

### ISSUE THREAD

**Issue by jschuler** [patternfly/patternfly-react#5205]
Copying from conversation with @karelhala, @jenny-s51 and @mcoker in slack:
> While I was going trough the component itself I think it could be improved, I don't think we need to include the hasSearchInput prop at all, we could allow passing custom components as consumers require, here is an example - https://codesandbox.io/s/cocky-yonath-65nuf?file=/index.js

we could provide more components to control that level, i.e. by adding a MenuInput and MenuContent container
```
<Menu>
  <MenuInput>
    <SearchInput />
  </MenuInput>
  <Divider>
  <MenuContent>
    <MenuList>
      <MenuListItem />
      <MenuListItem />
    </MenuList>
  </MenuContent>
</Menu>
```

Related: https://github.com/patternfly/patternfly-react/issues/5153

**Comment by karelhala** [comment]
Just a note that this should support multiple `Inputs` and spread around the menu itself, for instance to support searching in group
```
<Menu onSelect={this.onSelect} activeItemId={activeItem}>
  <MenuGroup>
    <MenuInput>
      <SearchInput />
    </MenuInput>
    <MenuContent>
      <MenuList>
      <MenuList>
    </MenuContent>
  </MenuGroup>
  <MenuGroup>
    <MenuInput>
      <SearchInput />
    </MenuInput>
    <MenuContent>
      <MenuList>
      <MenuList>
    </MenuContent>
  </MenuGroup>
</Menu>
```

Ideally it would be really good if we could have inputs as `MenuItem`s as well, but I would mark that as a stretch goal and might not actually be needed.

Also, while doing this change, I would reommend removing the `hasSearchInput` altogether. I'd be more happy if we didn't have to do some conditions to render searches and instead give consumers direct message that they can use whatever input they want to.

**Comment by jschuler** [comment]
@mcoker is that something we can already with with the Core CSS? Add inputs at the group level

**Comment by mcoker** [comment]
Technically you can, there are no styles associated with `<MenuGroup>` or `<MenuContent>`.

<img width="1254" alt="Screen Shot 2020-12-03 at 10 36 59 AM" src="https://user-images.githubusercontent.com/35148959/101059302-baf35900-3553-11eb-8a15-c7323b7ca710.png">

cc @mcarrano

**Comment by jenny-s51** [comment]
@mcarrano what are your thoughts on composing the menu component this way? @mcoker and I are curious to hear your thoughts

**Comment by tlabaj** [comment]
closed by #5211
